pkg/manager/manager.go

```diff
@@ -79,6 +79,12 @@ type Manager interface {
 	// AddReadyzCheck allows you to add Readyz checker
 	AddReadyzCheck(name string, check healthz.Checker) error
 
+	// AddPrestartHook allows you to add a hook that runs after leader election and immediately
+	// before starting controllers needing leader election. Prestart hooks block execution of
+	// leader election controllers until all return nil error. The manager is stopped on non-nil
+	// errors.
+	AddPrestartHook(func(ctx context.Context) error) error
```

Comment:
Lets make this take a `Runnable` to stay consistent

Reply:
I thought about this, but decided against it. While the types are the same (so Go would be happy) the meaning isn't. The hook functions do some work then exit, with the manager blocked until all return with non-nil error However, the meaning of a Runnable is different:

> Start blocks until the context is closed or an error occurs.

Reply:
Runnables can also exit cleanly without having the manager fail, so that should be fine?

Reply:
That is not the reading I have of this GoDoc.

Reply:
The comment I guess might be out of date, but runnables can exit with a `nil` error

Reply:
Okay. I wish I was told that back in November. :sweat_smile: 

I can include rephrasing in this CL, or we can handle it in another one.